src/main/java/org/terasology/climateConditions/visualization/ShowMapCommand.java

```diff
@@ -16,48 +16,102 @@
+import static org.terasology.climateConditions.ConditionsBaseField.TEMPERATURE;
+
+@Share(ShowMapCommand.class)
 @RegisterSystem
 public class ShowMapCommand extends BaseComponentSystem {
-    public static final int SIZE_OF_IMAGE = 300;
-
-    @In
-    ClimateMapDisplaySystem climateSystem;
+    public static final int SIZE_OF_IMAGE = 30;
 
     @In
     private ClimateConditionsSystem climateConditions;
-
+    @In
+    private LocalPlayer localPlayer;
     @In
     private NUIManager nuiManager;
+    @In
+    private WorldProvider worldProvider;
+
+    private String type;
+    private int mapHeight;
+
+    private void prepareBases(String mapType) {
+        mapHeight = (int) localPlayer.getPosition().y;
+
+        if (climateConditions.getWorldSeed() == null) {
+            setClimateSeed();
+        }
+        if (climateConditions.getTemperatureBaseField() == null) {
+            climateConditions.configureTemperature();
+        }
+        if (climateConditions.getHumidityBaseField() == null) {
+            climateConditions.configureHumidity();
+        }
+
+        if (mapType.equals(TEMPERATURE)) {
+            type = TEMPERATURE;
+        } else {
+            type = HUMIDITY;
+        }
+    }
 
     /**
      * Displays the selected map at the selected height level.
      *
      * The lighter a pixel on the map, the higher temperature/humidity/whatever else.
      * @param mapType
      */
-    @Command(shortDescription = "Display condition map", helpText = "Show a given map (humidity, temperature) at a given height level.")
-    public void showClimateMap(@CommandParam("map type") String mapType, @CommandParam("height to look at") int height) {
-        climateSystem.setMapHeight(height);
+    @Command(shortDescription = "Display condition map", helpText = "Show a given map (humidity, temperature) around the player.")
+    public void showClimateMap(@CommandParam("map type") String mapType) {
+        prepareBases(mapType);
+
+        nuiManager.pushScreen("ClimateConditions:displayConditionScreen");
+    }
 
-        if (mapType.equals("temperature")) {
-            climateSystem.setClimateConditionsBase(climateConditions.getTemperatureBaseField());
+    /**
+     * Gives the current temperature/humidity/etc. level at the player's position.
+     *
+     * @param mapType
+     * @return condition value
+     */
+    @Command(shortDescription = "Give condition value", helpText = "Gives the temperature/humidity/etc. of the player at this moment.")
+    public float climateValue(@CommandParam("map type") String mapType) {
+        prepareBases(mapType);
+
+        if (type.equals(TEMPERATURE)) {
+            return climateConditions.getTemperature(localPlayer.getPosition().x, localPlayer.getPosition().y, localPlayer.getPosition().z);
         } else {
-            climateSystem.setClimateConditionsBase(climateConditions.getHumidityBaseField());
+            return -1000;
```

Comment:
What does this mean?

Reply:
I second that question o.O

Reply:
That was just a mistake 😅
I fixed it in the latest commit.